### PR TITLE
This closes issue #53

### PR DESCRIPTION
**Pull request by msitu**
The error message in the first comment of #53 is fixed by using the correct type,`ICommMsgMsg`.

`karma-typescript-es6-transform` is needed to fix the "source-reader.karma-typescript" error that shows up when `npm test` is run. Adding that dependency necessitates updating some other dependencies related to Karma.

Here is more discussion of the need for the karma es6 transform:
- https://github.com/monounity/karma-typescript/issues/344#issue-463339619
- https://github.com/jupyter-widgets/widget-ts-cookiecutter/issues/53#issuecomment-543143885

### CODE REVIEW

**Comment by martinRenou** [comment]
Thanks for your fix! I tried locally, the tests compiles, but I get the following runtime error:
```
Firefox 72.0.0 (Ubuntu 0.0.0) ERROR
  ReferenceError: regeneratorRuntime is not defined
  at /tmp/karma-typescript-bundle-6867jBrYQlGE5uLT.js:114:30
  
  global.wrappers["/home/martin/github/martinRenou/ipygg/node_modules/d3-array/dist/d3-array.js"]</<@/tmp/karma-typescript-bundle-6867jBrYQlGE5uLT.js:114:30
  global.wrappers["/home/martin/github/martinRenou/ipygg/node_modules/d3-array/dist/d3-array.js"]</<@/tmp/karma-typescript-bundle-6867jBrYQlGE5uLT.js:110:117
  global.wrappers["/home/martin/github/martinRenou/ipygg/node_modules/d3-array/dist/d3-array.js"]<@/tmp/karma-typescript-bundle-6867jBrYQlGE5uLT.js:111:3
  require@node_modules/karma-typescript/dist/client/commonjs.js:17:25
  require/<@node_modules/karma-typescript/dist/client/commonjs.js:18:24
  global.wrappers["/home/martin/github/martinRenou/ipygg/src/scales/ColorUtils.ts"]<@src/scales/ColorUtils.ts:16:1 <- src/scales/ColorUtils.js:20:34
  require@node_modules/karma-typescript/dist/client/commonjs.js:17:25
  require/<@node_modules/karma-typescript/dist/client/commonjs.js:18:24
  global.wrappers["/home/martin/github/martinRenou/ipygg/src/scales/index.ts"]<@src/scales/index.ts:5:1 <- src/scales/index.js:36:22
  require@node_modules/karma-typescript/dist/client/commonjs.js:17:25
```

I wonder if we should not include `babel-polyfill` as well, see https://stackoverflow.com/questions/28976748/regeneratorruntime-is-not-defined.

Could you actually make it work with your PR?

**Comment by martinRenou** [comment]
Indeed, adding `babel-polyfill` fixed my issue

File: {{cookiecutter.github_project_name}}/package.json [hunk 1]
```diff
@@ -49,33 +49,34 @@
...
+    "@phosphor/application": "^1.7.3",
+    "@phosphor/widgets": "^1.9.3",
     "@types/expect.js": "^0.3.29",
-    "@types/mocha": "^5.2.5",
-    "@types/node": "^10.11.6",
-    "@types/webpack-env": "^1.13.6",
+    "@types/mocha": "^7.0.1",
+    "@types/node": "^13.7.0",
+    "@types/webpack-env": "^1.15.1",
     "css-loader": "^3.2.0",
```
Comment (martinRenou) [inline {{cookiecutter.github_project_name}}/package.json]: We might want to add babel polyfill

```suggestion
    "css-loader": "^3.2.0",
    "babel-polyfill": "^6.26.0",
```

File: {{cookiecutter.github_project_name}}/tests/karma.conf.js [hunk 2]
```diff
@@ -27,6 +27,14 @@ module.exports = function (config) {
```
Comment (martinRenou) [inline {{cookiecutter.github_project_name}}/tests/karma.conf.js]: And line 12 of this file, we need to include:

```
    files: [
      'node_modules/babel-polyfill/dist/polyfill.js',
      { pattern: "tests/src/**/*.ts" },
      { pattern: "src/**/*.ts" },
    ],
```

Comment (msitu) [reply {{cookiecutter.github_project_name}}/tests/karma.conf.js]: Whoops! Thanks for double checking. I just added babel-polyfill and confirmed locally that the tests pass and `jupyter lab` properly shows the sample widget.

**Comment by vidartf** [comment]
Note, the polyfills shouldn't really be needed, as we require modern browsers. Instead, share that information with the transpiler like this:
https://github.com/vidartf/opinionated-widget-cookiecutter/blob/b94f8a16c55a5f0014b2cfe41a0d0836b0fa469f/%7B%7Bcookiecutter.github_project_name%7D%7D/ts/tests/karma.conf.js#L51-L67

**Comment by msitu** [comment]
> Note, the polyfills shouldn't really be needed, as we require modern browsers. Instead, share that information with the transpiler like this:
> https://github.com/vidartf/opinionated-widget-cookiecutter/blob/b94f8a16c55a5f0014b2cfe41a0d0836b0fa469f/%7B%7Bcookiecutter.github_project_name%7D%7D/ts/tests/karma.conf.js#L51-L67

@vidartf, I add the extra config to the bundleOptions in karma.conf.js, and removed the babel-polyfill. Everything looks good on my end. Please confirm.

File: {{cookiecutter.github_project_name}}/package.json [hunk 3]
```diff
@@ -49,33 +49,34 @@
     "watch:nbextension": "webpack --watch"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1.1.10 || ^2"
+    "@jupyter-widgets/base": "^2.0.2"
```
Comment (vidartf) [inline {{cookiecutter.github_project_name}}/package.json]: Is this needed for the tests to pass?

**Comment by msitu** [comment]
When I run the sample app it no longer shows up as a blue node which black text when I print the value of `w`, and instead just prints `ExampleWidget()`. I will have to go back to the drawing board on this and will close this PR for now.

**Comment by vidartf** [comment]
@msitu There are other things wrong with the code base as well. That does not mean this PR is faulty. Does master work for you?

**Comment by wlievens** [comment]
Is there any progress on this issue?

**Comment by loichuder** [comment]
PR #58 appears to solve #53 and the rest for me but it is closed.

Would it be worthwhile to reopen it, verify and merge it ?

**Comment by wlievens** [comment]
As far as I'm concerned it works.

**Pull request closed without merging**